stubs/babel/babel/dates.pyi

```diff
@@ -1,30 +1,98 @@
+    format: Literal["narrow", "short", "long"] = ...,
+    locale: str | Locale = ...,
+) -> str: ...
+def format_skeleton(
+    skeleton: str,
+    datetime: time | datetime | None = ...,
+    tzinfo: tzinfo | None = ...,
+    fuzzy: bool = ...,
+    locale: str | Locale = ...,
+) -> str: ...
+def format_interval(
+    start: datetime | date | time,
+    end: datetime | date | time,
+    skeleton: str | None = ...,
+    tzinfo: tzinfo | None = ...,
+    fuzzy: bool = ...,
+    locale: str | Locale = ...,
+) -> str: ...
 
-def get_timezone(zone: Any | None = ...): ...
-def get_next_timezone_transition(zone: Any | None = ..., dt: Any | None = ...): ...
+# Timezone Functionality
+
+@overload
+def get_timezone(zone: str | BaseTzInfo | None = ...) -> BaseTzInfo: ...
+@overload
+def get_timezone(zone: tzinfo) -> tzinfo: ...
+def get_timezone_gmt(
+    datetime: datetime | None = ...,
+    width: Literal["long", "short", "iso8601", "iso8601_short"] = ...,
+    locale: str | Locale = ...,
+    return_z: bool = ...,
+) -> str: ...
+def get_timezone_location(
+    dt_or_tzinfo: datetime | tzinfo | None = ..., locale: str | Locale = ..., return_city: bool = ...
+) -> str: ...
+def get_timezone_name(
+    dt_or_tzinfo: datetime | tzinfo | None = ...,
+    width: Literal["long", "short"] = ...,
+    uncommon: bool = ...,
+    locale: str | Locale = ...,
+    zone_variant: Literal["generic", "daylight", "standard"] | None = ...,
+    return_zone: bool = ...,
+) -> str: ...
+
+# Note: While Babel accepts any tzinfo for the most part, the get_next_timeout_transition()
+# function requires a tzinfo that is produced by get_timezone()/pytz AND has DST info.
+# The typing here will help you with the first requirement, but will not protect against
+# pytz tzinfo's without DST info, like what you get from get_timezone("UTC") for instance.
+def get_next_timezone_transition(zone: BaseTzInfo | None = ..., dt: datetime | date | None = ...) -> TimezoneTransition: ...
 
 class TimezoneTransition:
-    activates: Any
-    from_tzinfo: Any
-    to_tzinfo: Any
-    reference_date: Any
-    def __init__(self, activates, from_tzinfo, to_tzinfo, reference_date: Any | None = ...) -> None: ...
+    # This class itself is not included in the documentation, yet it is mentioned by name.
+    # See https://github.com/python-babel/babel/issues/823
+    activates: datetime
+    from_tzinfo: tzinfo
+    to_tzinfo: tzinfo
+    reference_date: datetime | None
+    def __init__(
+        self, activates: datetime, from_tzinfo: tzinfo, to_tzinfo: tzinfo, reference_date: datetime | None = ...
+    ) -> None: ...
     @property
-    def from_tz(self): ...
+    def from_tz(self) -> str: ...
     @property
-    def to_tz(self): ...
+    def to_tz(self) -> str: ...
     @property
-    def from_offset(self): ...
+    def from_offset(self) -> int: ...
     @property
-    def to_offset(self): ...
+    def to_offset(self) -> int: ...
+
+UTC: BaseTzInfo
+LOCALTZ: BaseTzInfo
```

Comment:
These are re-exports from `babel.util` in the implementation, so they should be here too.